actions/restore.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+from charms import layer
+from charmhelpers.core.hookenv import action_fail
+from charmhelpers.core.hookenv import action_get
+from charmhelpers.core.hookenv import action_set
+from charmhelpers.core.hookenv import config
+from charmhelpers.core.hookenv import log
+from charmhelpers.core.hookenv import unit_get
+from charmhelpers.core.hookenv import resource_get
+from charmhelpers.core.host import chdir
+from charmhelpers.core.host import service_start
+from charmhelpers.core.host import service_stop
+from shlex import split
+from subprocess import check_call
+from subprocess import check_output
+from subprocess import Popen
+from subprocess import PIPE
+from datetime import datetime
+import hashlib
+import os
+import sys
+import time
+
+opts = layer.options('etcd')
+
+DATESTAMP = datetime.strftime(datetime.now(), '%Y%m%d-%H%M%S')
+ARCHIVE = "etcd-data-{}.tar.gz".format(DATESTAMP)
+
+ETCD_DATA_DIR = opts['etcd_data_dir']
+ETCD_PORT = config('management_port')
+PRIVATE_ADDRESS = unit_get('private-address')
+SKIP_BACKUP = action_get('skip-backup')
+SNAPSHOT_ARCHIVE = resource_get('snapshot')
+TARGET_PATH = action_get('target')
+
+
+def preflight_check():
```

Comment:
Since we are going to snaps do you need to add checks if you can write in the "etcd_data_dir" or something of that nature? I see that is a layer option and if someone did not know about snaps or confinement they may try to configure the directory incorrectly.